content/guides/concepts/hashes.md

```diff
@@ -0,0 +1,42 @@
+---
+title: "Hashes"
+menu:
+    guides:
+        parent: concepts
+---
+
+Hashes are functions that take some arbitrary input and return a fixed-size alphanumeric string. The appearance of that alphanumeric string depends on the given hash algorithm in use, such as [SHA-1](https://en.wikipedia.org/wiki/SHA-1) (used by Git), [SHA-256](https://en.wikipedia.org/wiki/SHA-2), or [BLAKE2](https://en.wikipedia.org/wiki/BLAKE_(hash_function)#BLAKE2). Have a look at the [full list of hash functions](https://en.wikipedia.org/wiki/List_of_hash_functions) for more.
+
+As an example, the input of the string:
+
+```
+Hello world
+```
+
+would be represented by **SHA-1** as:
+
+```
+0x7B502C3A1F48C8609AE212CDFB639DEE39673F5E
+```
+
+However, exactly the same input generates the following output by **SHA-256**:
+
+```
+0x64EC88CA00B268E5BA1A35678A1B5316D212F4F366B2477232534A8AECA37F3C
+```
+
+Notice that the second hash is longer than the first one. This is because SHA-1 creates a 160 bit hash, while SHA-256 creates a 256 bit hash. Also, the prepended `0x` is just and indicator that tells us that the following digest is represented as a hexadecimal number.
+
+Hashes can be represented in different bases (`base2`, `base16`, `base32`, etc.). In fact, IPFS makes use of that as part of its [Content Identifiers](cid.md) and supports mulitiple base representations at the same time, using the [Multibase](https://github.com/multiformats/multibase) protocol.
+
+## Characteristics of cryptographic hashes
+
+Cryptographic hashes come with a couple of very important characteristics:
+
+- **deterministic** - the same input message always returns exactly the same output hash
+- **uncorrelated** - a small change in the message should generate a completely different hash
+- **fast** - hashes are quickly computed
```

Comment:
Yeah, bcrypt is a good counter-example. It’s ideal for its use-case specifically because it’s *not* fast 😛

We should just drop this point. I’m guessing you pulled this list from Wikipedia’s cryptographic hash page, @PascalPrecht? It looks similar. (I wonder if I should go edit that page…)

Reply:
Yes I did take that from Wikipedia as I couldn't remember the characteristics I read in a book a while ago :) Will update.